spec.md

```diff
@@ -49,6 +49,119 @@ The following will not be part of the specification:
+   specification will be rendered by a network gateway on behalf of the
+   producer.
+
+2) Applications consume events originating from devices and/or people
+   and/or applications/-modules/-instances for display, archival, analytics,
+   workflow processing, or other handling.
+
+   Platform tools and applications consume events for the purpose of
+   monitoring the condition and/or providing transparency into the operation
+   of a business solution and its foundational building blocks.
+
+   The consumer application might reside on server hosts or on a device
+   of any kind.
+
+   A consuming application will typically be interested in:
+   - distinguishing events such that the exact same event is not
+     processed twice.
+   - identifying and selecting the origin context or the
+     producer-assigned classification.
+   - identifying the temporal order of the events relative to the
+     originating context and/or relative to a wall-clock.
+   - understanding the context-related detail information carried
+     in the event.
+
+   In some cases, the consuming application might be interested in:
+   - obtaining further details about the event's subject from the
+     originating context, like obtaining detail information about a
+     changed object that requires privileged access authorization.
+   - interact with the event's subject at the originating context,
+     for instance reading a storage blob after having been informed
+     that this blob has just been created.
+
+   Consumer interests motivate requirements for which information
+   producers ought to include an event.
+
+3) Middleware routes events from producers to consumers, or onwards
+   to other middleware. Applications producing events might delegate
+   certain tasks arising from their consumers' requirements to
+   middleware:
+
+   - Management of many concurrent interested consumers for one of
+     multiple classes or originating contexts of events
+   - Processing of filter conditions over a class or originating context
+     of events on behalf of consumers.
+   - Transcoding, like encoding in MsgPack after decoding from JSON
+   - Transformation that changes the event's structure, like mapping from
+     a proprietary format to Cloud Events, while preserving the
+     identity and semantic integrity of the event.
+   - Instant "push-style" delivery to interested consumers.
+   - Storing events for eventual delivery, either for pick-up initiated
+     by the consumer ("pull"), or initiated by the middleware ("push")
+     after a delay.
+   - Observing event content or event flow for monitoring or
+     diagnostics purposes.
+
+   To satisfy these needs, middleware will be interested in:
+   - A metadata discriminator usable for classification of events so that
+     consumers can express interest in one or multiple such classes.
+   - A metadata discriminator that allows distinguishing the subject of
+     a particular event of that class.
```

Comment:
what does "subject" mean in this context?

Reply:
in today's calls @clemensv clarified that this might be a specific file that is being acted on, which led to the event being generated.  I think this needs just a little more explanation.

Reply:
Agree with Sara. I think we need to define/explain what is meant by the subject. We may also need to define/explain classification of events. Classification can be based on different event source type (eg. storage, message notification etc.) or based on different cloud event vendor. What is this classification?